docs/docs/reference/changed-features/operators.md

```diff
@@ -0,0 +1,118 @@
+
+## The @alpha Annotation
+
+An `@alpha` annotation on a method definition defines an alternate name for the implementation of that method: Example:
+```scala
+object VecOps {
+  @alpha("append") def (xs: Vec[T]) ++= [T] (ys: Vec[T]): Vec[T] = ...
+}
+```
+Here, the `++=` operation is implemented (in Byte code or native code) under the name `append`. The implementation name affects the code that is generated, and is the name under which code from other languages can call the method. For instance, `++=` could be invoked from Java like this:
+```
+VecOps.append(vec1, vec2)
+```
+The `@alpha` annotation has no bearing on Scala usages. Any application of that method in Scala has to use `++=`, not `append`.
+
+An `@alpha` annotation will be _mandatory_ if the method name is symbolic. Symbolic methods without `@alpha` annotations are deprecated.
+
+### Motivation
+
+The `@alpha` annotation serves a dual purpose:
+
+ - It helps interoperability between Scala and other languages.
+ - It serves as a documentation tool by providing an alternative regular name
+   as an alias of a symbolic operator.
+
+### Details
+
+ 1. `@alpha` is defined in package `scala.annotation`. It takes a single argument
+    of type `String`. That string is called the _external name_ of the definition
+    that's annotated.
+
+ 2. An `@alpha` annotation can be given for all kinds of definitions.
+
+ 3. The name given in an `@alpha` annotation must be a legal name
+    for the defined entities on the host platform.
+
+ 4. Definitions with symbolic names should have an `@alpha` annotation. Lack of such
+    an annotation will raise a deprecation warning.
+
+ 5. Definitions with names in backticks that are not legal host platform names
+    should have an `@alpha` annotation. Lack of such an annotation will raise a deprecation warning.
+
+ 6. @alpha annotations must agree: If two definitions are members of an object or class with the same name and matching types, then either none of them has an `@alpha` annotation, or both have `@alpha` annotations with the same name.
+
+ 7. There must be a one-to-one relationship between external and internal names:
+ If two definitions are members of an object or class with matching types and both have `@alpha` annotations with the same external name, then their internal method names must also be the same.
+
+## The @infix Annotation
+
+An `@infix` annotation on a method definition allows using the method as an infix operation. Example:
+```scala
+trait MultiSet[T] {
+
+  @infix
+  def union(other: MultiSet[T]): MultiSet[T]
+
+  def difference(other: MultiSet[T]): MultiSet[T]
+
+  @alpha("intersection")
+  def *(other: MultiSet[T]): MultiSet[T]
+}
+
+val s1, s2: MultiSet[Int]
+
+s1 union s2         // OK
+s1.union(s2)        // also OK
+
+s1.difference(s2)   // OK
+s1 `difference` s2  // OK
+s1 difference s2    // gives a deprecation warning
+
+s1 * s2             // OK
+s1.*(s2)            // also OK, but unusual
```

Comment:
The two "also OK" cases (`s1.union(s2)` and `s1.*(s2)`) kind of go against the motivation of achieving consistency. For consistency to be really applied, they should give a deprecation warning.

Reply:
Yes, it's a tradeoff. I believe though that declaring something @infix would be a strong enough social cue that people would follow it without the compiler having to enforce it. Always allowing method call syntax has the advantage that we can normalize programs. We can say
`x union y` desugars to `x.union(y)`. If `x.union(y)` was not legal that explanation would be not available.

Reply:
Meh. Yes, OK.